modules/shared/src/main/scala/org/tessellation/schema/peer.scala

```diff
@@ -44,6 +44,12 @@ object peer {
       fromId(publicKey.toId)
   }
 
+  type WithPeerId[A] = Function1[A, PeerId]
```

Comment:
It doens't bring much of functionality imo. I would go with simple `_.map(_.id)` function each time when set of peers need to be verified. If you want to be fancy maybe a syntax would be enough, like:
```scala
val peers: List[Peers] = ...
peers.map(_.id).filterCollateralized(collateral)
```